### ISSUE TITLE
Seek assitance

### ISSUE THREAD

**Issue by kuang23** [SangHunHan92/2K2K#20]
Hi,

Here is my training setting of phase1 & phase2

image: shading image of Thuman2.0 + 2K2K dataset
lr: 1e-5
epoch: 30

I turn off background augmentation and image_blur in ReconDataset.py (With background augmentation, the model predicts black down normal maps during training in phase1. So I turn them off as you mentioned in #11 ) And the setting of colorjitter in transform_human in fetch_data is same as the default.

After the training of phase2, most of the predictions are not the ideal result.
To make the inference comparable, I change the colorjitter from (0.85,0.85) to (0.7,1.0) which is same as training, but the point cloud of one of training images is like 
<img width="333" alt="withcolorjitter" src="https://github.com/SangHunHan92/2K2K/assets/69189129/9bc58ee2-64b1-445f-9823-fffb1291ac03">

If I turn off the colorjitter, the result is better, but some points on left chest are abnormal
<img width="299" alt="withoutcolorjitter" src="https://github.com/SangHunHan92/2K2K/assets/69189129/2b219882-1226-4062-80ef-792f89e1c843">

I think these points are related to the region in red circle of the depth map during training.
<img width="549" alt="depthmap" src="https://github.com/SangHunHan92/2K2K/assets/69189129/acbaf04a-d1e3-4bd1-8673-5824b20a23fb">

So I am confused by two questions:
(1) Why does the result of turning colorjitter off seems better than using the training setting of colorjitter?
(2) What causes the inaccurate depth prediction in the red circle of depth map?

I will appreciate for your suggestions and expertise.

**Comment by SangHunHan92** [comment]
(1) I added color jitter to make it robust to real images such as YouTube videos or DSLR cameras. But if test images don't contain various shade, colorjitter is meaningless.

(2) I have never encountered this problem.
Please check the Normal map or Image that affects the depth map first. 
If both are correct, it may be the network has not learned enough.
I recommend to freeze the normal network that has been finished training and train only the depth network from scratch.

**Comment by kuang23** [comment]
Thanks for your sharing!!!

I think these two problem are probably caused by BatchNorm2D.

(1) I change model.eval to model.train in test_02_model to make BatchNorm2D track the mean and var of test data. The result becomes normal.

(2) I also convert BatchNorm2D to syncBatchNorm2D during training with multigpu. The deth map becomes better.

By the way, I wonder why you set the depth of background to 6 and apply relu6? Why not set to 0?

**Comment by SangHunHan92** [comment]
Thank you for the important information !!

I trained the front depth background to 6.
The reason is that the front depth map should have larger depth values as it approaches the boundary to ensure continuity with the back depth map.

However, if the background of the front depth map is trained to have a depth of 0, the depth value increases as it approaches the boundary of the depth map, but the value becomes abnormally decreases(to 0) at the boundary.

To prevent this, I set the average depth of the front and back to around 3 as the output of the network, and set the background value of the front depth to 6.
